File: c++.list.c
//3.7.4 list 大小操作
//功能描述：
//对list容器的大小进行操作
//函数原型：
//size();                 //返回容器中元素的个数
//empty();                //判断容器是否为空
//resize(num);            //重新指定容器的长度为num，若容器变长，则以默认值填充新位置。
//					    //如果容器变短，则末尾超出容器长度的元素被删除。
//resize(num, elem);       //重新指定容器的长度为num，若容器变长，则以elem值填充新位置。
//​					    //如果容器变短，则末尾超出容器长度的元素被删除。

#include<iostream>
using namespace std;
#include<list>
void printList(const list<int>&L){
	for (list<int>::const_iterator it = L.begin(); it != L.end(); it++){
		cout << *it << " ";
	}
	cout << endl;
}
void test01(){
	list<int>L1;
	L1.push_back(10);
	L1.push_back(20);
	L1.push_back(30);
	L1.push_back(40);
	printList(L1);
	//判断容器是否为空
	if (L1.empty()){
		cout << "L1为空" << endl;
	}
	else{
		cout << "L1不为空" << endl;
		cout << "L的元素个数为：" << L1.size() << endl;
	}
	L1.resize(10,520);
	printList(L1);
	L1.resize(1);
	printList(L1);
}
int main(){
	test01();
	return 0;
}
//总结：
//- 判断是否为空-- - empty
//- 返回元素个数-- - size
//- 重新指定个数-- - resize

File: c++.list.f
//3.7.7 list 反转和排序
功能描述：
将容器中的元素反转，以及将容器中的数据进行排序
函数原型：**
reverse();    //反转链表
sort();         //链表排序
#include<iostream>
using namespace std;
#include<list>
#include<algorithm>
void printList(const list<int>&L){
	for (list<int>::const_iterator it = L.begin(); it != L.end(); it++){
		cout << *it << " ";
	}
	cout << endl;
}
void test01(){
	list<int>L;
	L.push_back(10); 
	L.push_back(50);
	L.push_back(30);
	L.push_back(40);
	cout << "反转前：" << endl;
	printList(L);
	//反转
	L.reverse();
	cout << "反转后：" << endl;
	printList(L);
}
bool myCompare(int v1,int v2){
	//降序 让第一个数 > 第二个数
	return v1 > v2;
}
void test02(){
	list<int>L;
	L.push_back(10);
	L.push_back(50);
	L.push_back(30);
	L.push_back(40);
	//排序
	cout << "排序前：" << endl;
	printList(L);
	//所有不支持随机访问迭代器的容器，不可以用便准算法
	//不支持随机访问迭代器的容器，内部会提供一些算法
	//sort(L.begin(), L.end());
	L.sort();//默认排序规则  从小到大 升序
	cout << "排序后：" << endl;
	printList(L);
	//降序
	L.sort(myCompare);
	printList(L);
}
int main(){
	test01();
	test02();
	return 0;
}
//总结：
//* 反转-- - reverse
//* 排序-- - sort （成员函数）
